Source/EventAnalysis/KTTrackProcessingWeightedSlope.cc

```diff
@@ -0,0 +1,250 @@
+    // Method associated with the <SWFC> slot
+    {
+
+        TrackID trackID = ExtractTrackID(tlcData);
+        Points& points = tlcData.GetPoints();
+        // The & is important!!
+        KTProcessedTrackData& procTrack = tlcData.template Of< KTProcessedTrackData >();
+
+        KTDEBUG(tlog, "Setting track reconstruction using \"weighted-slope\" algorithm");
+        return KTTrackProcessingWeightedSlope::DoWeightedSlopeAlgorithm(points, trackID, &procTrack);
+    }
+
+    bool KTTrackProcessingWeightedSlope::DoWeightedSlopeAlgorithm(Points& points, TrackID trackID,KTProcessedTrackData* procTrack)
+    {
+        vector< double > timeBinInAcq;
+        vector< double > timeBinInRunC;
+        vector< double > averageFrequency;
+
+        // Makes a first loop over the points to calculate the weighted average in one time slice
+        for (Points::const_iterator pIt = points.begin(); pIt != points.end(); ++pIt)
+        {
+            timeBinInAcq.push_back(pIt->fTimeInAcq);
+            timeBinInRunC.push_back(pIt->fTimeInRunC);
+        }
+        //Make these lists duplicate-free
+        sort( timeBinInAcq.begin(), timeBinInAcq.end() );
+        timeBinInAcq.erase( unique( timeBinInAcq.begin(), timeBinInAcq.end() ), timeBinInAcq.end() );
+
+        sort( timeBinInRunC.begin(), timeBinInRunC.end() );
+        timeBinInRunC.erase( unique( timeBinInRunC.begin(), timeBinInRunC.end() ), timeBinInRunC.end() );
+
+        const int nTimeBins = timeBinInAcq.size();
+        //Derived Quantites
+        vector< double > mean(nTimeBins);
+        vector< double > variance(nTimeBins);
+        vector< double > neighborhoodAmplitude(nTimeBins);
+        vector< double > sumPf(nTimeBins);
+        vector< double > sumP(nTimeBins);
+        vector< double > trackSNR(nTimeBins);
+        vector< double > trackNUP(nTimeBins);
+        vector< double > wideTrackSNR(nTimeBins);
+        vector< double > wideTrackNUP(nTimeBins);
+        int nTrackBins = 0.;
+        // Calculate the averaged points
+        for (Points::const_iterator pIt = points.begin(); pIt != points.end(); ++pIt)
+        {
+            for (int iTimeBin=0; iTimeBin<nTimeBins; ++iTimeBin)
+            {
+                if (pIt->fTimeInAcq == timeBinInAcq[iTimeBin])
+                {
+                    sumPf[iTimeBin] += pIt->fFrequency * pIt->fAmplitude;
+                    sumP[iTimeBin] += pIt->fAmplitude;
+                    mean[iTimeBin] += pIt->fMean;
+                    variance[iTimeBin] += pIt->fVariance;
+                    neighborhoodAmplitude[iTimeBin] += pIt->fNeighborhoodAmplitude;
+
+                    trackSNR[iTimeBin] += pIt->fAmplitude/pIt->fMean;
+                    trackNUP[iTimeBin] += (pIt->fAmplitude - pIt->fMean) / pIt->fVariance;
+
+                    wideTrackSNR[iTimeBin] += pIt->fNeighborhoodAmplitude/pIt->fMean;
+                    wideTrackNUP[iTimeBin] += (pIt->fNeighborhoodAmplitude - pIt->fMean) / pIt->fVariance;
```

Comment:
NUP += residual / sqrt(pIt->fVariance)?

Reply:
You are right, but I don't think I did this piece though. I can modify this and solve it before merging